AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/CredentialStorage/AWSCognitoAuthCredentialStore.swift

```diff
@@ -90,6 +91,12 @@ struct AWSCognitoAuthCredentialStore {
         return "\(storeKey(for: authConfiguration)).\(sessionKey)"
     }
 
+    private func generateDeviceMetadataKey(
+        for username: String,
+        with configuration: AuthConfiguration) -> String {
+            return "\(storeKey(for: authConfiguration)).\(username).\(deviceMetadataKey)"
```

Comment:
We are passing `AuthConfiguration` inside the string? Can you show an example on how this look like?

Reply:
This is not a string... its passed in a method to create a unique key.. 

```
    private func storeKey(for authConfiguration: AuthConfiguration) -> String {
        let prefix = "amplify"
        var suffix = ""

        switch authConfiguration {
        case .userPools(let userPoolConfigurationData):
            suffix = userPoolConfigurationData.poolId
        case .identityPools(let identityPoolConfigurationData):
            suffix = identityPoolConfigurationData.poolId
        case .userPoolsAndIdentityPools(let userPoolConfigurationData, let identityPoolConfigurationData):
            suffix = "\(userPoolConfigurationData.poolId).\(identityPoolConfigurationData.poolId)"
        }

        return "\(prefix).\(suffix)"
    }
```